frappe/website/doctype/personal_data_delete_request/personal_data_delete_request.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _
+from frappe.model.document import Document
+from frappe.utils.verified_command import get_signed_params, verify_request
+
+class PersonalDataDeleteRequest(Document):
+
+	def after_insert(self):
+		if frappe.session.user in ['Administrator', 'Guest']:
+			frappe.throw(_("This user's data cannot be requested for deletion"))
+		else:
+			self.send_verification_mail()
+
+	def send_verification_mail(self):
+		url = frappe.utils.get_url("/api/method/frappe.website.doctype.personal_data_delete_request.personal_data_delete_request.confirm_deletion") +\
+		"?" + get_signed_params({"email": self.email, "name": self.name})
+
+		frappe.sendmail(recipients= self.email,
+		subject=_("ERPNext: Confirm Deletion of Data"),
+		template="delete_data_confirmation",
+		args={'email':self.email, 'link':url},
+		header=[_("ERPNext: Confirm Deletion of Data"), "green"])
+
+	def anonymize_data(self):
```

Comment:
validate permission here

Reply:
Added server-side validation for anonymize data.